### PR TITLE
Fix binding of fully qualified view reference

### PR DESCRIPTION
**Pull request by Tishj**
This PR fixes the issue mentioned in #3724 

When using a fully qualified column reference (schema.table.column) with a View as the table part, the Binder will fail to resolve this, because the View is bound as a regular Binding, which does not have any link to a schema.

I've fixed this issue by adding an optional schema string to the Binding, that can now be used to verify if a Binding is part of a schema or not, when relevant.

One potential problem I foresee with this change is that this now uses the binding alias, instead of the name from the CatalogEntry inside Binder::HasMatchingBinding.
I'll try to write a test that will highlight this problem, if it is indeed a problem.

If it is a problem, I think we might need to create another class derived from Binding that holds a CatalogEntry, which the TableBinding could also inherit from, then it won't have to rely on the LogicalGet for the catalog entry either.

### CODE REVIEW

**Comment by Tishj** [comment]
Ah.. as I expected, it is a problem
`test_multi_column_reference.test:45` unexpectedly succeeds now

**Comment by Tishj** [comment]
The first implementation turned out to cause a problem, so instead I've opted to create a Binding that contains a reference to a StandardEntry, that the View now makes use of. Now `GetStandardEntry` (formerly named `GetTableEntry`) now returns the ViewCatalogEntry and the schema name can be verified in `Binder::HasMatchingBinding()`

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Looks good to me. Some minor nitpicks:

File: src/include/duckdb/planner/table_binding.hpp [hunk 1]
```diff
@@ -24,8 +24,9 @@ class LogicalGet;
 class TableCatalogEntry;
 class TableFunctionCatalogEntry;
 class BoundTableFunction;
+class StandardEntry;
 
-enum class BindingType { BASE, TABLE, MACRO };
+enum class BindingType { BASE, TABLE, MACRO, ENTRY };
```
Comment (Mytherin) [inline src/include/duckdb/planner/table_binding.hpp:29]: `CATALOG_ENTRY` maybe?

**Review by Mytherin — COMMENTED** [review]
One more nitpick :)

**Pull request merged**